cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1221,6 +1226,8 @@ func mountPartitionMatchingKernelDisk(dir, fallbacklabel string) error {
 		// first partition we will be mounting, we can't know if anything is
 		// corrupted yet
 		NeedsFsck: true,
+		// don't need nosuid option here, since this function is only used
+		// for ubuntu-boot and ubuntu-seed, never ubuntu-data
```

Comment:
maybe we should then rename the function to mountNonDataPartitionMathingKernelDisk?